common/remote/rmtsmtp.cpp

```diff
@@ -907,7 +903,25 @@ static void doSendEmail(CMailInfo & info, CMailPart const & part)
         StringBuffer outbuff;
 
         info.read(0);
-        info.getEhloAndOptions(outbuff.clear());
+
+        // Try to use ESMTP and fall back to SMTP if necessary
+        try
+        {
+            info.getEhlo(outbuff.clear());
+            info.writeAndAck(outbuff.str(), outbuff.length());
+            info.processEsmtpOptions(outbuff);
+        }
+        catch(IException * e)
+        {
+            // Log error as a warning
+            StringBuffer msg;
+            info.addToWarnings(e->errorMessage(msg).str());
+            EXCLOG(MCoperatorError, e, "WARNING");
```

Comment:
MCoperatorWarning?

Reply:
I copied that from somewhere. Do you have a better way of doing this?